static/src/javascripts/projects/common/modules/commercial/track-ad.js

```diff
@@ -0,0 +1,30 @@
+define([
+    'Promise',
+    'common/utils/mediator'
+], function(Promise, mediator) {
+    return trackAd;
+
+    function trackAd(id) {
+        return new Promise(function (resolve, reject) {
+            var onAdLoaded = function (event) {
+                if (event.slot.getSlotElementId() === id) {
```

Comment:
I guess it depends if we want to let modules listen to all rendering events (unless mediator handles wildcard listeners?)

The next step is to get rid of mediator altogether. Knowing if an ad has rendered is a stateful query, events are stateless and transient.